### PR TITLE
feature/se-43

### LINKED ISSUE #43: Create the front end for the Quiz
image

### PR DESCRIPTION
**Pull request by j0world**
close [SE-WS23/documentation#43 ](https://github.com/SE2-WS23/documentation/issues/43)

86BC4C00-ACCF-4AEE-B067-36A97DB7594E
implements the quiz page 
currently works on dummy data, next button doesn't do anything yet

### CODE REVIEW

File: src/main.jsx [hunk 1]
```diff
@@ -46,6 +47,11 @@ const router = createBrowserRouter([
     element: <AssessmentPage />,
     errorElement: <ErrorPage />,
   },
+  {
+    path: "/quiz",
+    element: <QuizPage />,
+    errorElement: <ErrorPage />,
+  },
```
Comment (julix14) [inline src/main.jsx]: Please merge the current main branch into yours, there was a bigger change in the router and i think you have to adapt your code to it

File: src/pages/Quiz/QuizPage.jsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        <PageLayout title="Quiz" viewportPage>
+    
+            <Box
+                sx={{
+                flexGrow: 1,
+                display: "flex",
+                flexDirection: "column",
+                justifyContent: "center",
+                alignItems: "center",
+                height: "calc(100% - 60px)",
```
Comment (julix14) [inline src/pages/Quiz/QuizPage.jsx:70]: Why did you calculate the hight here?

File: src/pages/Quiz/QuizPage.jsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+/**
+ * Renders the Assessment page.
+ * @returns {JSX.Element} QuizPage component
+ */
+
+function QuizPage() {
+
+    const [currentQuestion, setcurrentQuestion] = useState(0);
+    const [progressValue, setprogressValue] = useState(1);
```
Comment (julix14) [inline src/pages/Quiz/QuizPage.jsx]:
```suggestion
    const [currentQuestion, setCurrentQuestion] = useState(0);
    const [progressValue, setProgressValue] = useState(1);
```

**Review by julix14 — CHANGES_REQUESTED** [review]
image
I think you messed up the layout a bit. If you change the screenheight a bit it looks like this

File: src/pages/Quiz/QuizPage.jsx
Comment (Chrissydrx) [inline src/pages/Quiz/QuizPage.jsx:1]: Just as a general question: Should we implement an implementation if a question is answered correct or wrong?

File: src/pages/Quiz/QuizPage.jsx (hunk 3 above)
Comment (Chrissydrx) [inline src/pages/Quiz/QuizPage.jsx]: Why is the default value for the progress Bar 1?

File: src/pages/Quiz/QuizPage.jsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    const handleAnswerButtonClick = (answer) => {
+        if (quiztions[currentQuestion].correct === answer.target.textContent) // checks if the answer is correct
+            console.log("Correct");
+        else
+            console.log("Incorrect");
+
+        if (currentQuestion !== quiztions.length - 1) { // updates questions and progress bar
+            setcurrentQuestion(currentQuestion + 1);
+            setprogressValue((((currentQuestion +1)/ quiztions.length) * 100));
+        }
```
Comment (Chrissydrx) [inline src/pages/Quiz/QuizPage.jsx]:
```suggestion
        if (currentQuestion !== quiztions.length - 1) { // updates questions and progress bar
            setcurrentQuestion(currentQuestion + 1);
            setprogressValue((((currentQuestion +1)/ quiztions.length) * 100));
            return;
        }
```

File: src/pages/Quiz/QuizPage.jsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    const [currentQuestion, setcurrentQuestion] = useState(0);
+    const [progressValue, setprogressValue] = useState(1);
+
+    const handleAnswerButtonClick = (answer) => {
+        if (quiztions[currentQuestion].correct === answer.target.textContent) // checks if the answer is correct
+            console.log("Correct");
+        else
+            console.log("Incorrect");
+
+        if (currentQuestion !== quiztions.length - 1) { // updates questions and progress bar
```
Comment (Chrissydrx) [inline src/pages/Quiz/QuizPage.jsx]: The comment is miss leading and is not necessary

File: src/pages/Quiz/QuizPage.jsx [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+            setcurrentQuestion(currentQuestion + 1);
+            setprogressValue((((currentQuestion +1)/ quiztions.length) * 100));
+        }
+
+        // update the progress bar on the last question
+        if (currentQuestion === quiztions.length-1)
+            setprogressValue(100); 
+    }
+
+    const quiztions = [ // placeholder for the real questions
```
Comment (Chrissydrx) [inline src/pages/Quiz/QuizPage.jsx]: What are quiztions? Do you mean questions?
Maybe you should name it dummyQuestions

File: src/pages/Quiz/QuizPage.jsx [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+                display: "flex",
+                flexDirection: "column",
+                justifyContent: "center",
+                alignItems: "center",
+                height: "calc(100% - 60px)",
+                }}
+            >
+                <LinearProgress 
+                    sx={{
+                    margin: "4% 0 3% 0",
```
Comment (Chrissydrx) [inline src/pages/Quiz/QuizPage.jsx]: Margin in percent? I think, this can lead to unwanted effects

File: src/pages/Quiz/QuizPage.jsx [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+                    }} 
+                    variant="determinate" 
+                    value={progressValue} 
+                /> 
+
+                <Box
+                    sx={{
+                    height: "80%",
+                    width: "80%",
+                    backgroundColor: "#D9D9D9",
```
Comment (Chrissydrx) [inline src/pages/Quiz/QuizPage.jsx]: Please you theme data for colors

File: src/pages/Quiz/QuizPage.jsx [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+                    flexGrow: 1,
+                    display: "flex",
+                    flexDirection: "column",
+                    justifyContent: "center",
+                    alignItems: "center",
+                    }}
+                >
+                    
+                    <h1 style={{margin: "0% 0 15% 0"}}>
```
Comment (Chrissydrx) [inline src/pages/Quiz/QuizPage.jsx]: Margin in percent? I think, this can lead to unwanted effects

File: src/pages/Quiz/QuizPage.jsx [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+                        {quiztions[currentQuestion].question}
+                    </h1>
+
+                    {quiztions[currentQuestion].answers?.map((answer) => (
+                        <Button
+                            variant="contained"
+                            color="primary"
+                            size="large"
+                            sx={{ width: "70%", margin: "2% 0"}}
```
Comment (Chrissydrx) [inline src/pages/Quiz/QuizPage.jsx]: Margin in percent? I think, this can lead to unwanted effects

File: src/pages/Quiz/QuizPage.jsx (hunk 2 above)
Comment (Chrissydrx) [inline src/pages/Quiz/QuizPage.jsx:70]: The height overlaps with the button

File: src/pages/Quiz/QuizPage.jsx
Comment (Chrissydrx) [inline src/pages/Quiz/QuizPage.jsx:1]: image
The page is not responsive

**Review by Chrissydrx — CHANGES_REQUESTED** [review]

File: src/pages/Quiz/QuizPage.jsx (hunk 3 above)
Comment (j0world) [reply src/pages/Quiz/QuizPage.jsx]: It was the value I had in there when it finally started working. It also sets a minimal amount of progress even before answering the first question and I thought that looked cuter :)

Comment (j0world) [reply src/pages/Quiz/QuizPage.jsx]: done

File: src/pages/Quiz/QuizPage.jsx (hunk 5 above)
Comment (j0world) [reply src/pages/Quiz/QuizPage.jsx]: deleted

File: src/pages/Quiz/QuizPage.jsx (hunk 4 above)
Comment (j0world) [reply src/pages/Quiz/QuizPage.jsx]: done

File: src/pages/Quiz/QuizPage.jsx (hunk 6 above)
Comment (j0world) [reply src/pages/Quiz/QuizPage.jsx]: I thought it was a funny wordplay. It is changed now

File: src/pages/Quiz/QuizPage.jsx (hunk 2 above)
Comment (j0world) [reply src/pages/Quiz/QuizPage.jsx:70]: Honestly, idk. Copied from assessment page for uniformity

File: src/pages/Quiz/QuizPage.jsx (hunk 7 above)
Comment (j0world) [reply src/pages/Quiz/QuizPage.jsx]: yep, I just learned that vw and vh are a thing :)
please check if it is better now

File: src/pages/Quiz/QuizPage.jsx (hunk 8 above)
Comment (j0world) [reply src/pages/Quiz/QuizPage.jsx]: I don't quite understand. The grey tone is the same as in the Refined Wireframe and the assessment page. Should I import the theme from somewhere?

Comment (Chrissydrx) [reply src/pages/Quiz/QuizPage.jsx]: The assessment page is right now only a placeholder. So, it is not really part of the theme. The Background color of the Quiz will be final and should reflect the theme. So, please you theme data

File: src/pages/Quiz/QuizPage.jsx (hunk 3 above)
Comment (Chrissydrx) [reply src/pages/Quiz/QuizPage.jsx]: Why does it not work with 0?

File: src/pages/Quiz/QuizPage.jsx (hunk 6 above)
Comment (Chrissydrx) [reply src/pages/Quiz/QuizPage.jsx]: Please don't use wordplays in coding. You variables should be self-explanatory and nobody should guess what you mean with it.

**Review by Chrissydrx — CHANGES_REQUESTED** [review]

File: src/pages/Quiz/QuizPage.jsx (hunk 3 above)
Comment (j0world) [reply src/pages/Quiz/QuizPage.jsx]: it does work with zero, I just think it looks better with one

File: src/pages/Quiz/QuizPage.jsx (hunk 8 above)
Comment (j0world) [reply src/pages/Quiz/QuizPage.jsx]: The background color is just the same as in the wire frame and I thought that was fine. 
As I still don't quite understand, would you mind rephrasing what I need to do?
Is the grey box also supposed to turn Blue like the buttons?

Comment (j0world) [reply src/pages/Quiz/QuizPage.jsx]: Screenshot 2023-11-13 at 19 51 26
:(

I have yet to push this and would like clarification whether this is what you meant.

Comment (j0world) [reply src/pages/Quiz/QuizPage.jsx]: Screenshot 2023-11-15 at 15 25 31


Quiz with the new light blue, please say if this is ok

File: src/pages/Quiz/QuizPage.jsx (hunk 9 above)
Comment (j0world) [reply src/pages/Quiz/QuizPage.jsx]: changed to vh/ vw

File: src/pages/Quiz/QuizPage.jsx (hunk 10 above)
Comment (j0world) [reply src/pages/Quiz/QuizPage.jsx]: changed to vh/ vw

File: src/pages/Quiz/QuizPage.jsx
Comment (j0world) [reply src/pages/Quiz/QuizPage.jsx:1]: hopefully fixed, it's better now for sure

**Comment by Chrissydrx** [comment]
Resolve Changes and create new PR

**Pull request closed without merging**